build_projects/update-dependencies/Config.cs

```diff
@@ -73,5 +74,38 @@ private static string GetEnvironmentVariable(string name, string defaultValue =
 
             return value;
         }
+
+        private static string GetDefaultUpstreamBranch()
+        {
+            return GetRepoMSBuildPropValue("BranchInfo.props", "BranchName") ?? "master";
+        }
+
+        private static string GetDefaultCoreSetupVersionFragment()
+        {
+            string coreSetupChannel = GetRepoMSBuildPropValue("BundledRuntimes.props", "CoreSetupChannel") ?? "master";
+
+            return $"dotnet/core-setup/{coreSetupChannel}";
+        }
+
+        private static string GetRepoMSBuildPropValue(string propsFileName, string propertyName)
+        {
+            var propsFilePath = Path.Combine(Dirs.RepoRoot, "build", propsFileName);
+            var root = XDocument.Load(propsFilePath).Root;
+            var ns = root.Name.Namespace;
+
+            var value = root
```

Comment:
> Nit: unclear vars.

@livarcocc loves it this way.

> Also, does it seem reasonable to change this into an msbuild task to avoid the parsing? (Maybe longer term.)

It's 5 lines of parsing...  but yea maybe someday